File: starter_kits/C++/hlt/dropoff.cpp
#include "dropoff.hpp"
#include "input.hpp"

std::shared_ptr<hlt::Dropoff> hlt::Dropoff::_generate(hlt::PlayerId player_id) {
    hlt::EntityId dropoff_id;
    int x;
    int y;
    hlt::get_sstream() >> dropoff_id >> x >> y;

    return std::make_shared<hlt::Dropoff>(player_id, dropoff_id, x, y);
}

File: starter_kits/Rust/src/hlt/dropoff.rs
use hlt::DropoffId;
use hlt::entity::Entity;
use hlt::input::Input;
use hlt::PlayerId;
use hlt::position::Position;

pub struct Dropoff {
    pub owner: PlayerId,
    pub id: DropoffId,
    pub position: Position,
}

impl Entity for Dropoff {
    fn owner(&self) -> PlayerId {
        self.owner
    }

    fn position(&self) -> Position {
        self.position
    }
}

impl Dropoff {
    pub fn generate(input: &mut Input, player_id: PlayerId) -> Dropoff {
        input.read_and_parse_line();
        let id = DropoffId(input.next_usize());
        let x = input.next_i32();
        let y = input.next_i32();

        Dropoff { owner: player_id, id, position: Position { x, y } }
    }
}
